vowpalwabbit/text_parser/include/vw/text_parser/parse_example_text.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "vw/common/string_view.h"
+#include "vw/core/multi_ex.h"
+#include "vw/core/vw_fwd.h"
+
+#include <cstdint>
+
+namespace VW
+{
+namespace parsers
+{
+namespace text
+{
+void substring_to_example(VW::workspace* all, VW::example* ae, VW::string_view example);
+
+void read_line(VW::workspace& all, example* ex, VW::string_view line);  // read example from the line.
+void read_line(VW::workspace& all, example* ex, const char* line);      // read example from the line.
```

Comment:
We should remove this one and just require using the string view overload

Reply:
plain remove or deprecate? I'm fine with plain remove

Reply:
We don't need to deprecate it because this file is brand new

Reply:
I meant from the original file

Reply:
The approach I have been taking is if the thing moves to a non-details namespace then I leave a compatibility func or alias. If it moves to details I don't worry about that. So in this case since it isn't details leaving the func in is safer

Reply:
yeah I agree I just removed

Reply:
made a string_view version of read_lines too while we are at it